compiler/shared/main/scala/mlscript/compiler/codegen/CppAst.scala

```diff
@@ -0,0 +1,198 @@
+package mlscript.compiler.codegen.cpp
+
+import mlscript._
+import mlscript.utils._
+import mlscript.utils.shorthands._
+import mlscript.compiler.utils._
+
+enum Specifier:
+  case Extern
+  case Static
+  case Inline
+
+  def toDocument = this match
+    case Extern => "extern" |> raw 
+    case Static => "static" |> raw
+    case Inline => "inline" |> raw
+
+  override def toString(): Str = toDocument.print
+
+object Type:
+  def toDocuments(args: Ls[Type], sep: Document, extraTypename: Bool = false): Document =
+    args.zipWithIndex.map {
```

Comment:
When transforming a list just for the purpose of folding it later, better use an iterator to avoid intermediate allocations.
```suggestion
    args.iterator.zipWithIndex.map {
```
The same remark applies everywhere else you do this, like in the next function.